midi_app_controller/controller/connected_controller.py

```diff
@@ -0,0 +1,363 @@
+        action_handler.
+
+
+        Parameters
+        ----------
+        data : List[int]
+            Standard MIDI message.
+        """
+        pass  # TODO: for now we're not handling button disengagement
+
+    def handle_knob_message(self, data):
+        """Runs the action bound to the knob turn, specified in
+        action_handler.
+
+
+        Parameters
+        ----------
+        data : List[int]
+            Standard MIDI message.
+        """
+        id = data[0]
+        velocity = data[1]
+        prev_velocity = self.knob_engagement[id]
+
+        if velocity == self.controller.knob_value_min:
+            prev_velocity = self.controller.knob_value_min + 1
+
+        if velocity == self.controller.knob_value_max:
+            prev_velocity = self.controller.knob_value_max - 1
+
+        self.action_handler.handle_knob_action(
+            knob_id=id,
+            old_value=prev_velocity,
+            new_value=velocity,
+        )
+
+    def send_midi_message(self, data):
+        """Sends the specified MIDI message, using python-rtmidi
+        MidiIn class.
+
+        Parameters
+        ----------
+        data : List[int]
+            Standard MIDI message.
+        """
+        try:
+            self.midi_out.send_message(data)
+        except ValueError as err:
+            print(f"Value Error: {err}")
+
+    def flash_knob(self, id):
+        """Flashed the LEDs corresponding to a knob on a
+        MIDI controller.
+
+        Parameters
+        ----------
+        position : int
+            Position of the knob.
+        """
+        sleep_seconds = 0.3
+
+        for _ in range(3):
+            self.change_knob_value(id, self.controller.knob_value_min)
+            time.sleep(sleep_seconds)
+            self.change_knob_value(id, self.controller.knob_value_max)
+            time.sleep(sleep_seconds)
+
+    def flash_button(self, id):
+        """Flashed the button LED on a MIDI controller.
+
+        Parameters
+        ----------
+        position : int
+            Position of the button.
+        """
+        for _ in range(3):
+            self.turn_on_button_led(id)
+            time.sleep(0.3)
+            self.turn_off_button_led(id)
+            time.sleep(0.3)
+
+    def change_knob_value(self, id, new_value):
+        """Sends the MIDI message, responsible for changing
+        a value assigned to a knob.
+
+        Parameters
+        ----------
+        id : int
+            Knob id.
+        new_value : int
+            Value to set the knob to.
+        """
+        data = [ControllerConstants.knob_value_change_command, id, new_value]
+
+        self.send_midi_message(data)
+
+    def turn_on_button_led(self, id):
+        """Sends the MIDI message, responsible for changing
+        the button LED to 'on' state.
+
+        Parameters
+        ----------
+        id : int
+            Button id.
+        """
+        data = [
+            ControllerConstants.button_value_change_on_command,
```

Comment:
Please do not hide the channel number in the constants. Be explicit that we use a specific single channel for now. Maybe add a function that will build the `data` based on `id`, `value`, `message type` and `channel`.